images/win/scripts/Installers/Install-Msys2.ps1

```diff
@@ -88,4 +75,15 @@ pacman.exe -Q | grep -v ^mingw-w64-
 
 Write-Host "`nMSYS2 installation completed"
 
+# Environment
+# add C:\msys64\mingw64\bin and C:\msys64\usr\bin to PATH
```

Comment:
I'm not sure about adding both of these being a good idea. IIRC, the former corresponds to MINGW64 packages and the later to MSYS packages. Those are different terminals, and many packages might be duplicated.

Moreover, I don't think that adding any of these to the PATH is required. The shells defined in `Configure-Shell.ps1` (above) do need to be available in the PATH. But, then, users can enter the desired terminal, and the corresponding ecosystem will be available already. Adding these PATHs will expose tools which might not work in cmd/powershell due to missing libs/dlls.

/cc @lazka

Reply:
MSYS2 also add those two to PATH (among others..), mingw tools will win anyway.

Reply:
After removing C:\Program Files\Git\mingw64\bin and C:\Program Files\Git\usr\bin folders from PATH, binaries which are resolved from them, will become unavailable. This change can cause issues and impact since some customers / tools depend on those binaries. We decided to replace:
```
C:\Program Files\Git\usr\bin -> C:\msys64\usr\bin
C:\Program Files\Git\mingw64\bin -> C:\msys64\mingw64\bin
```

This folders contain pretty the same list of binaries like C:\Program Files\Git\usr\bin with minimal differences and mostly the same versions. If we add C:\msys64\usr\bin at the end of PATH, removing Git tools from PATH should be pretty smoothly since the most tools will continue to work as previously. (Unix tools that are resolved from System32 folder right now, will continue to work as expected since system32 is located before in PATH)

Reply:
@al-cheb, my point is that, after this PR is merged, users of `shell: sh`, `shell: bash`, `shell: msys2bash` or `shell: gitbash` will already have all those tools available, by default, without the need to add them explicitly. That's how MSYS2 shells work, as @lazka said.

Hence, adding these locations to the PATH is only useful for users of `shell: powershell` or `shell: cmd` which rely on unix tools. That feels like a corner case to me, specially given that unix tools from System32 are resolved first, and those are the ones that users of powershell/cmd should expect only.

However, my concern is with python, ruby, etc. which might be installed as "native" tools (by downloading installers from their websites) or through MSYS2's `pacman`. Adding MSYS2 locations to the PATH will automatically expose those tools. However, if used from powershell/cmd, chances are that e.g. shared libs will be missing. Hence, the expected behaviour is only guaranteed for statically compiled tools.

Should the concern be that a relevant subset of users expects some Unix tools to be available in powershell/cmd, but those don't exist in System32; then, I'd recommend symlinking those tools explicitly, instead of adding the location of all MSYS2/MINGW64 binaries. Otherwise, we are reproducing the mistake of exposing binaries out of context (i.e. not expected to be used that way).

Reply:
The problem is that `powershell` is default shell on Windows so the most of users use it by default.
For example, toolkit `which` function [searches through PATH entries to find tool](https://github.com/actions/toolkit/blob/a5e05630d8e1d89cd75b5a12a972daae8abc6e1a/packages/io/src/io.ts#L236). Some default actions like `actions/cache` (and probably, some custom tasks) depend on it.

I agree that it is not good to expose binaries out of context but since they are already exposed right now, we have to keep them for backward compatibility.
As for the option with symlinks, we don't really know what tools should be symlinked right now. Finally, we can come to point where we symlink all tools + maintenance concerns (hardcoded list of symlinked tools).

If we add MSYS folder at the end of PATH, it should safe enough because it will be resolved only in case when tool was not resolved from all other places.

Reply:
@maxim-lobanov, fair enough. Then, I would strongly suggest to have this clearly explained in the docs. If the toolkit or some other Actions rely on it, it's ok not to break backward compatibility. Yet, in the mid-long term, the those Actions should be updated. I believe the better approach is to start with making it explicit that this is a temporal corner case solution.

Reply:
Agree, @al-cheb could you please also update documentation to include info about it?